### PR TITLE
PWX-30496:  if '-T dmthin' exists in stc before preflight is ran and preflight fails don't start.   

### PR DESCRIPTION
**Pull request by jrivera-px**
**What this PR does / why we need it**:
If the user intended to use dmthin.    The '-T dmthin' will exist in the stc before preflight is ran.  If preflight fails in this case don't start.
**Which issue(s) this PR fixes** (optional)
Closes #
PWX-30496

**Special notes for your reviewer**:

### CODE REVIEW

File: drivers/storage/portworx/preflight.go [hunk 1]
```diff
@@ -180,8 +185,36 @@ func (u *preFlightPortworx) RunPreFlight() error {
...
+			}
+		}
+	}
+
+	// Check for pre-existing DMthin in misc-args
+	if mArgs, err := pxutil.MiscArgs(u.cluster); err == nil {
+		checkArgs(mArgs)
+	} else {
+		logrus.Warnf("runPreFlight: error parsing misc args, skipping existing PX-StoreV2 param check: %v", err)
+	}
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/preflight.go:203]: We are already checking the pod args below. Why do we need to check the cluster's misc args again?

File: drivers/storage/portworx/preflight.go [hunk 2]
```diff
@@ -245,9 +278,31 @@ func (u *preFlightPortworx) ProcessPreFlightResults(recorder record.EventRecorde
...
 	if passed {
-		// Enable DMthin via misc args
-		u.cluster.Annotations[pxutil.AnnotationMiscArgs] = strings.TrimSpace(u.cluster.Annotations[pxutil.AnnotationMiscArgs] + " -T dmthin")
-		k8sutil.InfoEvent(recorder, u.cluster, util.PassPreFlight, "Enabling DMthin")
+		if !u.hardFail { // Enable DMthin via misc args if not enabled already
+			u.cluster.Annotations[pxutil.AnnotationMiscArgs] = strings.TrimSpace(u.cluster.Annotations[pxutil.AnnotationMiscArgs] + " -T dmthin")
+			// Only set if CloudStorage is nil
+			if u.cluster.Spec.CloudStorage == nil {
+				// Add 64G metadata drive.
+				cmetadata := "type=gp2,size=64"
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/preflight.go:286]: We should use gp3 here.

File: drivers/storage/portworx/preflight.go [hunk 3]
```diff
@@ -245,9 +278,31 @@ func (u *preFlightPortworx) ProcessPreFlightResults(recorder record.EventRecorde
 	}
 
 	if passed {
-		// Enable DMthin via misc args
-		u.cluster.Annotations[pxutil.AnnotationMiscArgs] = strings.TrimSpace(u.cluster.Annotations[pxutil.AnnotationMiscArgs] + " -T dmthin")
-		k8sutil.InfoEvent(recorder, u.cluster, util.PassPreFlight, "Enabling DMthin")
+		if !u.hardFail { // Enable DMthin via misc args if not enabled already
+			u.cluster.Annotations[pxutil.AnnotationMiscArgs] = strings.TrimSpace(u.cluster.Annotations[pxutil.AnnotationMiscArgs] + " -T dmthin")
+			// Only set if CloudStorage is nil
+			if u.cluster.Spec.CloudStorage == nil {
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/preflight.go:284]: - Let's add the metadata device in both cases where the user has passed `-T dmthin` or we added it for them.
- We need to handle the case where cloud storage is not nil. 
- We should not overwrite the metadata device if already provided by the user.
```
if u.cluster.Spec.CloudStorage == nil {
  u.cluster.Spec.CloudStorage = &corev1.CloudStorageSpec{}
}

if u.cluster.Spec.CloudStorage.SystemMdDeviceSpec != nil {
  u.cluster.Spec.CloudStorage.SystemMdDeviceSpec = &cmetadata
}
```

File: drivers/storage/portworx/preflight.go (hunk 1 above)
Comment (jrivera-px) [reply drivers/storage/portworx/preflight.go:203]: Args can be added in the spec correct.  So we should check them both..

**Review by piyush-nimbalkar — APPROVED** [review]
Let's add the metadata device if the check passes irrespective if the user added the dmthin arg or not.

**Pull request merged**